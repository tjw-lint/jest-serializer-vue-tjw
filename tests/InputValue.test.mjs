const helpers = require('./helpers.js');

import { shallowMount } from '@vue/test-utils';
import InputValue from './components/InputValue.vue';

describe('InputValue.vue', () => {
  test('Value shown in snapshot', () => {
    helpers.mockSettings({ addInputValues: true });

    const wrapper = shallowMount(InputValue);

    expect(wrapper)
      .toMatchSnapshot();
  });

  test('Value not shown in snapshot', () => {
    helpers.mockSettings({ addInputValues: false });

    const wrapper = shallowMount(InputValue);

    expect(wrapper)
      .toMatchSnapshot();
  });
});
