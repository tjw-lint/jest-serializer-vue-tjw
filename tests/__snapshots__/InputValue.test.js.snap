// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputValue.vue Value not shown in snapshot 1`] = `
<form>
  <input value="Native value with attribute" title="text">
  <input value="Native value without attribute">
  <input value="Native checkbox" type="checkbox">
  <input title="text">
  <input>
  <input>
  <input>
  <input>
  <input>
  <input type="date">
  <input type="date">
  <input>
  <input>
  <input>
  <input>
  <input type="range">
  <input>
  <input>
  <input type="checkbox">
  <input>
  <input>
  <input>
  <textarea></textarea>
  <input type="radio" value="cat">
  <input type="radio" value="dog">
  <select>
    <option value="carrot"></option>
    <option value="celery"></option>
  </select>
</form>
`;

exports[`InputValue.vue Value shown in snapshot 1`] = `
<form>
  <input value="Native value with attribute" title="text">
  <input value="Native value without attribute">
  <input value="Native checkbox" type="checkbox">
  <input title="text" value="'There is a title attribute on this input'">
  <input value="'There are no other attrubutes on this input'">
  <input value="['text',22]">
  <input value="false">
  <input value="true">
  <input value="Error: err">
  <input type="date" value="1577836800000">
  <input type="date" value="Invalid Date">
  <input value="Infinity">
  <input value="-Infinity">
  <input value="NaN">
  <input value="null">
  <input type="range" value="75">
  <input value="{a:{c:true},b:{d:'text'}}">
  <input value="['a','b','c']">
  <input type="checkbox" value="'Text'">
  <input value="Function">
  <input value>
  <input value="undefined">
  <textarea value="'Text\\nText'"></textarea>
  <input type="radio" value="cat" v-model="'dog'">
  <input type="radio" value="dog" v-model="'dog'">
  <select value="'celery'">
    <option value="carrot"></option>
    <option value="celery"></option>
  </select>
</form>
`;
